diff --git a/drivers/media/i2c/techpoint/techpoint_tp2855.c b/drivers/media/i2c/techpoint/techpoint_tp2855.c
index 9d3c422d3..4ffb5c4b2 100644
--- a/drivers/media/i2c/techpoint/techpoint_tp2855.c
+++ b/drivers/media/i2c/techpoint/techpoint_tp2855.c
@@ -46,9 +46,11 @@ static __maybe_unused const struct regval common_setting_594M_1080p_regs[] = {
 	{ 0x14, 0x33 },
 	{ 0x14, 0xb3 },
 	{ 0x14, 0x33 },
+	{ 0x45, 0x01 },
 
 	// {0x23, 0x02}, //vi test ok
 	// {0x23, 0x00},
+	{0x40, 0x04},		//add this for all channel can be change
 };
 
 static __maybe_unused const struct regval common_setting_594M_720p_1chn_2lane_regs[] = {
@@ -346,7 +348,7 @@ int tp2855_set_channel_reso(struct i2c_client *client, int ch,
 		techpoint_write_reg(client, 0x2d, 0x54);
 		techpoint_write_reg(client, 0x2e, 0x40);
 		techpoint_write_reg(client, 0x30, 0xa5);
-		techpoint_write_reg(client, 0x31, 0x86);
+		techpoint_write_reg(client, 0x31, 0x82); //0x86
 		techpoint_write_reg(client, 0x32, 0xfb);
 		techpoint_write_reg(client, 0x33, 0x60);
 		break;
